### PR TITLE
Signing targets for binfmt, rngd, compilers, toybox, tini

### PR DESCRIPTION
**Pull request by riyazdf**
Set up notary delegations and Makefile `sign` targets for:
```
docker.io/linuxkit/binfmt
docker.io/linuxkit/rngd
docker.io/linuxkit/go-compile
docker.io/linuxkit/c-compile
docker.io/linuxkit/toybox-media
docker.io/linuxkit/tini
```

Also updated yml files and Makefiles to verify signatures where appropriate.

### CODE REVIEW

File: pkg/binfmt/Makefile [hunk 1]
```diff
@@ -23,6 +23,12 @@ push: tag
 	rm -f hash
 	docker rmi $(IMAGE):build || true
```
Comment (justincormack) [inline pkg/binfmt/Makefile]: this Makefile doesn't `pull` the base image, so is missing the trust check on that

**Comment by justincormack** [comment]
Not sure why CI is failing

File: pkg/binfmt/Makefile [hunk 2]
```diff
@@ -10,6 +11,7 @@ DEPS=Dockerfile Makefile main.go $(wildcard etc/binmft.d/*)
 # the hashing stuff soon anyway, so there is little point in faffing
 # about with computing a hash during the build and get it out.
 hash: $(DEPS)
+	DOCKER_CONTENT_TRUST=1 docker pull $(BASE)
```
Comment (justincormack) [inline pkg/binfmt/Makefile]: put this under `tag` where the build is done not here, it makes more sense.

File: pkg/rngd/Makefile [hunk 3]
```diff
@@ -2,10 +2,12 @@
 default: push
 
 IMAGE=rngd
+BASE=linuxkit/c-compile:f52f485825c890d581e82a62af6906c1d33d8e5d
 SHA_IMAGE=alpine:3.5@sha256:dfbd4a3a8ebca874ebd2474f044a0b33600d4523d03b0df76e5c5986cb02d7e8
 DEPS=Dockerfile Makefile
 
 hash: $(DEPS)
+	DOCKER_CONTENT_TRUST=1 docker pull $(BASE)
```
Comment (justincormack) [inline pkg/rngd/Makefile]: and here

File: Makefile [hunk 4]
```diff
@@ -18,13 +18,15 @@ endif
 PREFIX?=/usr/local/
 
 bin/moby: | bin
-	docker run --rm --log-driver=none $(CROSS) $(GO_COMPILE) --clone-path github.com/moby/tool --clone https://github.com/moby/tool.git --package github.com/moby/tool/cmd/moby --ldflags "-X main.GitCommit=$(GIT_COMMIT) -X main.Version=$(VERSION)" -o $@ > tmp_moby_bin.tar
+	DOCKER_CONTENT_TRUST=1 docker pull $(GO_COMPILE)
+	DOCKER_CONTENT_TRUST=1 docker run --rm --log-driver=none $(CROSS) $(GO_COMPILE) --clone-path github.com/moby/tool --clone https://github.com/moby/tool.git --package github.com/moby/tool/cmd/moby --ldflags "-X main.GitCommit=$(GIT_COMMIT) -X main.Version=$(VERSION)" -o $@ > tmp_moby_bin.tar
```
Comment (justincormack) [inline Makefile]: not sure we should set DCT for the tool builds, as it breaks ability to compile offline, and this is not part of the product. Users can do this themselves, but long term we will provide other ways to get the binaries that have signing, eg package managers etc.

File: Makefile [hunk 5]
```diff
@@ -18,13 +18,15 @@ endif
...
 	tar xf tmp_moby_bin.tar > $@
 	rm tmp_moby_bin.tar
 	touch $@
 
 LINUXKIT_DEPS=$(wildcard src/cmd/linuxkit/*.go) Makefile vendor.conf
 bin/linuxkit: $(LINUXKIT_DEPS) | bin
+	DOCKER_CONTENT_TRUST=1 docker pull $(GO_COMPILE)
```
Comment (justincormack) [inline Makefile]: and here

File: pkg/binfmt/Makefile [hunk 6]
```diff
@@ -14,7 +15,8 @@ hash: $(DEPS)
 
 tag: hash
 	docker pull linuxkit/$(IMAGE):$(shell cat hash) || \
```
Comment (justincormack) [inline pkg/binfmt/Makefile]: this pull should have DCT too? Then we replace unsigned version with signed one. If its not signed its not real!

Comment (justincormack) [inline pkg/binfmt/Makefile]: this should set DCT at least in the case we are doing `sign` else if there is an existing unsigned tag it wont be replaced. Maybe means splitting signed and unsigend cases though.

File: Makefile (hunk 4 above)
Comment (justincormack) [inline Makefile]: actually for `moby` you cant currently build offline i guess but for linuxkit you can.

File: pkg/binfmt/Makefile (hunk 6 above)
Comment (riyazdf) [inline pkg/binfmt/Makefile]: split out a `signed-tag` for all `sign` targets

File: pkg/binfmt/Makefile (hunk 1 above)
Comment (riyazdf) [inline pkg/binfmt/Makefile]: fixed

**Pull request merged**